src/libertem/udf/com.py

```diff
@@ -680,12 +677,22 @@ def get_results(self):
         field = np.moveaxis(np.array(field), 0, -1)
         nav_size = prod(self.meta.dataset_shape.nav)
 
-        regression, inp = self.get_regression(field)
+        # Only apply the regression to the valid
+        # part of the results:
+        valid_mask = self.meta.get_valid_nav_mask(full_nav=True).reshape(
+            tuple(self.meta.dataset_shape.nav)
+        )
+        regression, inp = self.get_regression(field, valid_mask=valid_mask)
+
         if inp is not None:
             self.apply_lin_regression(regression, inp, field)
         elif not np.allclose(regression[0], 0):
             self.apply_mean_regression(regression, field)
 
+        # zero the field values inside of the invalid-mask
+        # as it might be non-zero from the applied regression:
+        # field[np.broadcast_to(mask, field.shape)] = 0
```

Comment:
Why is this in there, commented out? The variable `mask` is not defined in the previous code, as far as I can tell.

Reply:
Went ahead and limited the application of regression to valid data.